src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1518,7 +1518,10 @@ private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOExcepti
         PrintWriter w = null;
         try {
             w = new PrintWriter(ssh);
-            w.println("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"");
+            // avoid echoing command as part of the password
+            w.println("@echo off");
+            // no need for quotes on windows echo -- they will get echoed too
+            w.println("echo " + Secret.toString(sshUser.getPassphrase()));
```

Comment:
When I created a private key using a passphrase which intentionally includes command processor special characters (like pipe, caret, less than, greater than, at sign, etc.), then try to use that passphrase ( `<mark|earl^waite>@gmail.com`) with the code, the initial attempt to authenticate with that repository (from the web user interface), reports:

    Failed to connect to repository : Command "C:\Program Files\git\bin\git.exe -c core.askpass=true ls-remote -h mark-pc1.markwaite.net:/var/lib/git/hasphrase/bin.git HEAD" returned status code 128:
    stdout: 
    stderr: 'earlwaite' is not recognized as an internal or external command, 
    operable program or batch file. 
    'earlwaite' is not recognized as an internal or external command, 
    operable program or batch file. 
    Permission denied, please try again. 
    'earlwaite' is not recognized as an internal or external command, 
    operable program or batch file. 
    Permission denied, please try again. 
    'earlwaite' is not recognized as an internal or external command, 
    operable program or batch file. 
    Permission denied (publickey,password). 
    fatal: Could not read from remote repository.
    
    Please make sure you have the correct access rights
    and the repository exists.